packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -78,146 +76,95 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
       return;
     }
     this.started = true;
-    if(!this.contract) {
-      await this.initContract();
+    await this.initialize();
+  }
+
+  /**
+   * Main methods to be called either to build genesis state or to save data to propose blocks later.
+   * @param isScanEth1ForGenesis
+   * @param fromBlock
+   */
+  public async getDepositEventsFromBlock(
```

Comment:
this name implicates it will fetch deposit event from single block. Better name could be "getDepositEventsSource(abortSignal, fromBlock?: number)". I don't think you need "isScanEth1ForGenesis", we should have separate method "getEth1BlocksSource"